checkmate/checker/url/url_haus.py

```diff
@@ -0,0 +1,80 @@
+"""URLHaus (https://urlhaus.abuse.ch/) based checker."""
+
+from tempfile import TemporaryDirectory
+
+from checkmate.checker.pipeline import Download, Pipeline, ReadCSVFile, UnzipFile
+from checkmate.checker.url.reason import Reason
+from checkmate.models import URLHausRule
+from checkmate.url import hash_for_rule
+
+
+class URLHaus:
+    """A checker which works against and updates URLHaus rules."""
+
+    # The columns provided in the CSV from URLHaus are:
+    # id, dateadded, url, url_status, threat, tags, urlhaus_link, reporter
+    COLUMN_ID = 0
+    COLUMN_URL = 2
+
+    INITIAL_FEED = Pipeline(
+        [
+            Download("https://urlhaus.abuse.ch/downloads/csv/"),
+            UnzipFile("csv.txt"),
+            ReadCSVFile(),
+        ]
+    )
+
+    UPDATE_FEED = Pipeline(
+        [
+            Download("https://urlhaus.abuse.ch/downloads/csv_recent/"),
+            ReadCSVFile(),
+        ]
+    )
```

Comment:
Here are the pipelines (and the pay off for the above). It was much nicer when I had like 12 of these to have them all expressed like this rather than having custom code for each one.

When called these will run through the pipeline and return the output of the last stage. In this case a generator of rows from the CSV.